report_viewer/apps/viewer/middleware/report_views_middleware.py

```diff
@@ -0,0 +1,75 @@
+"""report_views_middleware - logs views of the reports to the database"""
+import hashlib
+from uuid import UUID
+from report_viewer.settings.base import SECRET_KEY
+from accessibility_monitoring_platform.apps.s3_read_write.models import S3Report
+from accessibility_monitoring_platform.apps.reports.models import ReportVisitsMetrics
+
+
+class ReportMetrics:
+    """Middle for recording report views to the database"""
+
+    digit_to_animal_hash = [
+        "Owl",
+        "Dog",
+        "Cat",
+        "Mongoose",
+        "Chicken",
+        "Ferret",
+        "Rhino",
+        "Horse",
+        "Pig",
+        "Yak",
+    ]
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def client_ip(self, request) -> str:
+        x_forwarded_for = request.META.get("HTTP_X_FORWARDED_FOR")
+        if x_forwarded_for:
+            return x_forwarded_for.split(",")[0]
+        return request.META.get("REMOTE_ADDR")
+
+    def string_to_hash(self, request) -> str:
+        return (
+            request.META.get("HTTP_USER_AGENT", "")
+            + self.client_ip(request)
+            + SECRET_KEY
+        )
+
+    def fingerprint_hash(self, string_to_hash: str) -> int:
```

Comment:
Docstrings would be nice in this entire section to explain what is going on and why.

Reply:
added